utils/lister.cc

```diff
@@ -81,53 +81,54 @@ future<> lister::rmdir(fs::path dir) {
 }
 
 directory_lister::~directory_lister() {
-    if (_lister) {
+    if (_gen) {
         on_internal_error(llogger, "directory_lister not closed when destroyed");
     }
 }
 
 future<std::optional<directory_entry>> directory_lister::get() {
-    if (!_opt_done_fut) {
-        auto dir = co_await open_checked_directory(general_disk_error_handler, _dir.native());
-        auto walker = [this] (fs::path dir, directory_entry de) {
-            return _queue.push_eventually(std::make_optional<directory_entry>(std::move(de)));
-        };
-        SCYLLA_ASSERT(!_lister);
-        _lister = std::make_unique<lister>(std::move(dir), _type, std::move(walker), _filter, _dir, _do_show_hidden);
-        _opt_done_fut = _lister->done().then_wrapped([this] (future<> f) {
-            if (f.failed()) [[unlikely]] {
-                _queue.abort(f.get_exception());
-                return make_ready_future<>();
-            }
-            return _queue.push_eventually(std::nullopt);
-        }).finally([this] {
-            _lister.reset();
-        });
+    if (!_opened) {
+        _opened = co_await open_checked_directory(general_disk_error_handler, _dir.native());
+        _gen.emplace(_opened.experimental_list_directory());
+    }
```

Comment:
How about adding an async `open()` method and convert the callers?

Reply:
Possible, but it's "preexisting problem". Why do you think it's worth adding one? The lister is such since it was first introduced